src/Sentry.Unity/UnitySentryOptions.cs

```diff
@@ -12,32 +13,147 @@ public enum SentryUnityCompression
+        /// <summary>
+        /// Relative to Assets/Resources
+        /// </summary>
+        public const string ConfigRootFolder = "Sentry";
+        public const string ConfigName = "SentryOptions";
+
+        public bool Enabled { get; set; } = true;
+        public bool CaptureInEditor { get; set; } = true; // Lower entry barrier, likely set to false after initial setup.
+        public string? Dsn { get; set; }
+        public bool Debug { get; set; } = true; // By default on only
+        public bool DebugOnlyInEditor { get; set; } = true;
+        public SentryLevel DiagnosticsLevel { get; set; } = SentryLevel.Error; // By default logs out Error or higher.
         // Ideally this would be per platform
         // Auto allows us to try figure out things in the SDK depending on the platform. Any other value means an explicit user choice.
-        [field: SerializeField] public SentryUnityCompression RequestBodyCompressionLevel { get; set; } = SentryUnityCompression.Auto;
-        [field: SerializeField] public bool AttachStacktrace { get; set; }
-        [field: SerializeField] public float SampleRate { get; set; } = 1.0f;
+        public SentryUnityCompression RequestBodyCompressionLevel { get; set; } = SentryUnityCompression.Auto;
+        public bool AttachStacktrace { get; set; }
+        public float SampleRate { get; set; } = 1.0f;
 
-        [field: NonSerialized] public IDiagnosticLogger? Logger { get; private set; }
-        [field: NonSerialized] public string? Release { get; set; }
-        [field: NonSerialized] public string? Environment { get; set; }
+        public IDiagnosticLogger? Logger { get; private set; }
+        public string? Release { get; set; }
+        public string? Environment { get; set; }
 
-        public void OnEnable()
+        // Can't rely on Unity's OnEnable() hook.
+        public UnitySentryOptions TryAttachLogger()
         {
-            hideFlags = HideFlags.DontUnloadUnusedAsset;
             Logger = Debug
                      && (!DebugOnlyInEditor || Application.isEditor)
                 ? new UnityLogger(DiagnosticsLevel)
                 : null;
+
+            return this;
+        }
+
+        public void WriteTo(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+
+            writer.WriteBoolean("enabled", Enabled);
+            writer.WriteBoolean("captureInEditor", CaptureInEditor);
+
+            if (!string.IsNullOrWhiteSpace(Dsn))
+            {
+                writer.WriteString("dsn", Dsn);
+            }
+
+            writer.WriteBoolean("debug", Debug);
+            writer.WriteBoolean("debugOnlyInEditor", DebugOnlyInEditor);
+            writer.WriteString("diagnosticsLevel", DiagnosticsLevel.ToString().ToLowerInvariant());
+            writer.WriteString("requestBodyCompressionLevel", RequestBodyCompressionLevel.ToString());
```

Comment:
I think best is to serialize the numeric representation of all enums. It'll version better (if we add new enum values we can always deserialize it even if that value is't defined on the new enum.

```suggestion
            writer.WriteString("diagnosticsLevel", DiagnosticsLevel.ToString("d"));
            writer.WriteString("requestBodyCompressionLevel", RequestBodyCompressionLevel.ToString("d"));
```

Reply:
Yes, I was gonna ask you the same question, because this was my initial idea too.